src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -6752,6 +6752,31 @@ public void TestWin32ResWithBadResFile_CS1583ERR_BadWin32Res()
             CleanupAllGeneratedFiles(badres);
         }
 
+        [Fact(), WorkItem(217718, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=217718")]
+        public void TestWin32ResWithBadResFile_CS1583ERR_BadWin32Res_02()
+        {
+            string source = Temp.CreateFile(prefix: "", extension: ".cs").WriteAllText(@"class Test { static void Main() {} }").Path;
+            string badres = Temp.CreateFile().WriteAllBytes(new byte [] { 0, 0}).Path;
+
+            var baseDir = Path.GetDirectoryName(source);
+            var fileName = Path.GetFileName(source);
+
+            var outWriter = new StringWriter(CultureInfo.InvariantCulture);
+            int exitCode = new MockCSharpCompiler(null, baseDir, new[]
+            {
+                "/nologo",
+                "/preferreduilang:en",
+                "/win32res:" + badres,
+                source
+            }).Run(outWriter);
+
+            Assert.Equal(1, exitCode);
+            Assert.Equal("error CS1583: Error reading Win32 resources -- Unrecognized resource file format.", outWriter.ToString().Trim());
```

Comment:
I think multiple win32res parameters can be set. Should we include the path to the file in the error?